vane-proxy-core/src/main/java/org/oddlama/vane/proxycore/VaneProxyPlugin.java

```diff
@@ -0,0 +1,106 @@
+package org.oddlama.vane.proxycore;
+
+import org.jetbrains.annotations.NotNull;
+import org.oddlama.vane.proxycore.config.ConfigManager;
+import org.oddlama.vane.proxycore.config.IVaneProxyServerInfo;
+import org.oddlama.vane.proxycore.log.IVaneLogger;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.LinkedHashMap;
+import java.util.UUID;
+
+public abstract class VaneProxyPlugin {
+
+	public static final String CHANNEL_AUTH_MULTIPLEX = "vane_waterfall:auth_multiplex";
+	public final LinkedHashMap<UUID, UUID> multiplexedUUIDs = new LinkedHashMap<>();
+	public ConfigManager config = new ConfigManager(this);
+	public Maintenance maintenance = new Maintenance(this);
+	public IVaneLogger logger;
+	public ProxyServer server;
+
+	public boolean is_online(final IVaneProxyServerInfo server) {
+		final var addr = server.getSocketAddress();
+		if (!(addr instanceof final InetSocketAddress inet_addr)) {
+			return false;
+		}
+
+		var connected = false;
+		try (final var test = new Socket(inet_addr.getHostName(), inet_addr.getPort())) {
+			connected = test.isConnected();
+		} catch (IOException e) {
+			// Server not up or not reachable
+		}
+
+		return connected;
+	}
+
+	public String get_motd(final IVaneProxyServerInfo server) {
+		// Maintenance
+		if (maintenance.enabled()) {
+			return maintenance.format_message(Maintenance.MOTD);
+		}
+
+		final var cms = config.managed_servers.get(server.getName());
+		if (cms == null) {
+			return "";
+		}
+
+		String motd;
+		if (is_online(server)) {
+			motd = cms.motd(ManagedServer.MotdSource.ONLINE);
+		} else {
+			motd = cms.motd(ManagedServer.MotdSource.OFFLINE);
+		}
+
+		return motd;
+	}
+
+	public abstract java.io.File get_data_folder();
+
+	public abstract ProxyServer get_proxy();
+
+	public @NotNull IVaneLogger get_logger() {
+		return logger;
+	}
+
+	public @NotNull Maintenance get_maintenance() {
+		return this.maintenance;
+	}
+
+	public @NotNull ConfigManager get_config() {
+		return this.config;
+	}
+
+	public void try_start_server(ManagedServer server) {
+		// TODO: This could really use some checks (existing process running, nonzero exit code)
```

Comment:
I'm going to leave this for when I work on velocity.

Reply:
Very nice. I'm not sure whether we can reliably check for running processes, as the start command should not actually run the requested server's main process.

The command would block the proxy's main thread until it completes and would also interleave the processes output in the terminal. We should make it 100% clear that the command must only be a trigger that notifies another process.

For example a curl call to some API or removing a blockfile like the one used in the minecraft-server repository.

Reply:
Very nice. I'm not sure whether we can reliably check for running processes, as the start command should not actually run the requested server's main process.

The command would block the proxy's main thread until it completes and would also interleave the processes output in the terminal. We should make it 100% clear that the command must only be a trigger that notifies another process.

For example a curl call to some API or removing a blockfile like the one used in the minecraft-server repository.

Reply:
For "existing process" I meant the start command being executed. Right now if I have start script that takes awhile to run, the script will be rerun every time someone tries to join. What could be done is just keep the handle to the process and check `Process#isAlive` or setup something with `Process#onExit`. I haven't really looked into it enough to know what's best here yet.